processing/src/main/java/io/druid/segment/column/GenericColumn.java

```diff
@@ -39,16 +43,24 @@
   public String getStringSingleValueRow(int rowNum);
   @CalledFromHotLoop
   public Indexed<String> getStringMultiValueRow(int rowNum);
+
   @CalledFromHotLoop
-  public float getFloatSingleValueRow(int rowNum);
+  float getFloatSingleValueRow(int rowNum);
+  HistoricalFloatColumnSelector makeFloatSingleValueRowSelector(ReadableOffset offset);
```

Comment:
Should be FloatColumnSelector? or better to change the return types of the below methods because this class is used for only QueryableIndex?

Reply:
I didn't understand this. Why return types should be changed?

Reply:
For consistency. makeLongSingleValueRowSelector() and makeDoubleSingleValueRowSelector() return LongColumnSelector and DoubleColumnSelector, respectively.

Reply:
Also, please remove ```getXxxxMultiValueRow()``` methods because they are not used anymore.

Reply:
Interfaces like `HistoricalLongColumnSelector` and `HistoricalDoubleColumnSelector` don't exist yet. When they are added, the corresponding methods in GenericColumn and down the stack should be aligned with the current methods for the `float` specialization.

Reply:
Removed `getXxxxMultiValueRow()`